src/ILCompiler.WebAssembly/src/CodeGen/WebAssemblyObjectWriter.cs

```diff
@@ -690,6 +698,31 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
 
                     if (node.ShouldSkipEmittingObjectNode(factory))
                         continue;
+
+                    if (node is EETypeNode)
+                    {
+                        TypeDesc t = ((EETypeNode)node).Type;
+                        if (!t.IsArray)
```

Comment:
Why the test for array?

Reply:
byte[] comes through here and the subsequent call to factory.Vtable(t) dies with an assert.
``` 
Debug.Assert(!type.IsArray, "Wanted to call GetClosestDefType?");
```

Reply:
It might just work to make the line before this:

```csharp
DefType t = ((EETypeNode)node).Type.GetClosestDefType(); 
```

Whenever looking at or resolving virtual methods of non-pointer SzArray types, we need to switch over to looking at `System.Array<T>` type in System.Private.CoreLib. This has implementations of all the interesting virtuals (implementations of all the generic interfaces arrays implement).